src/Tribe/Attendees.php

```diff
@@ -752,72 +754,97 @@ public function sanitize_csv_value( $value ) {
-
-		// Verify event exists and current user has access to it.
-		if (
-			! $event instanceof WP_Post
-			|| ! $this->user_can( 'edit_posts', $event_id )
-		) {
+		if ( ! current_user_can( 'manage_options' ) ) {
 			return;
 		}
 
-		// Generate filtered list of attendees.
-		$items = $this->generate_filtered_list( $event_id );
+		if ( ! $all ) {
+			if ( empty( $_GET['event_id'] ) ) { // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+				return;
+			}
 
-		// Sanitize items for CSV usage.
-		$items = $this->sanitize_csv_rows( $items );
+			$event_id = absint( $_GET['event_id'] ); // phpcs:ignore WordPress.Security.NonceVerification.Recommended
 
-		/**
-		 * Allow for filtering and modifying the list of attendees that will be exported via CSV for a given event.
-		 *
-		 * @param array $items    The array of attendees that will be exported in this CSV file.
-		 * @param int   $event_id The ID of the event these attendees are associated with.
-		 */
-		$items = apply_filters( 'tribe_events_tickets_attendees_csv_items', $items, $event_id );
+			$event_id = Event::filter_event_id( $event_id, 'attendee-csv-report' );
+
+			// Verify event ID is a valid integer and the nonce is accepted.
+			if ( empty( $event_id ) || ! wp_verify_nonce( $_GET['attendees_csv_nonce'], 'attendees_csv_nonce' ) ) {
+				return;
+			}
+
+			$event = get_post( $event_id );
 
-		if ( ! empty( $items ) ) {
-			$charset  = get_option( 'blog_charset' );
-			$filename = sanitize_file_name( $event->post_title . '-' . __( 'attendees', 'event-tickets' ) );
+			// Verify event exists and current user has access to it.
+			if (
+				! $event instanceof WP_Post
+				|| ! $this->user_can( 'edit_posts', $event_id )
+			) {
+				return;
+			}
 
-			// Output headers so that the file is downloaded rather than displayed.
-			header( "Content-Type: text/csv; charset=$charset" );
-			header( "Content-Disposition: attachment; filename=$filename.csv" );
+			$items = $this->generate_filtered_list( $event_id );
+		} else {
+
+			$items = $this->generate_filtered_list( 'all' );
+		}
 
-			// Create the file pointer connected to the output stream.
-			$output = fopen( 'php://output', 'w' );
+		// Sanitize items for CSV usage.
+		$items = $this->sanitize_csv_rows( $items );
 
+		if ( ! $all ) {
 			/**
-			 * Allow filtering the field delimiter used in the CSV export file.
-			 *
-			 * @since 5.1.3
+			 * Allow for filtering and modifying the list of attendees that will be exported via CSV for a given event.
 			 *
-			 * @param string $delimiter The field delimiter used in the CSV export file.
+			 * @param array $items    The array of attendees that will be exported in this CSV file.
+			 * @param int   $event_id The ID of the event these attendees are associated with.
 			 */
-			$delimiter = apply_filters( 'tribe_tickets_attendees_csv_export_delimiter', ',' );
+			$items = apply_filters( 'tribe_events_tickets_attendees_csv_items', $items, $event_id );
```

Comment:
It feels odd to me to have this filter here instead of where the event items are generated. Or perhaps it feels more odd to have the filter only apply to single event items instead of all.

I understand that this reflects the usage before the change to allow `all` as an option, but perhaps we should consider making the filter apply regardless of whether one event or `all` has been set.

Reply:
Filters need docblcoks